app/controllers/application_controller.rb

```diff
@@ -83,7 +83,11 @@ def set_publication
   end
 
   def set_language_from_publication
-    I18n.locale = @publication.locale || I18n.default_locale
+    I18n.locale = if @publication.locale && I18n.available_locales.map(&:to_s).include?(@publication.locale)
```

Comment:
Might I suggest setting the Rails `I18n.available_locales` to the list of locales supported by the schema instead of adding this complexity to the controller?

The list is
```
[
  "ar", "az", "be", "bg", "bn", "cs", "cy", "de", "dr", "el", "en", "es", 
  "es-419", "et", "fa", "fr", "he", "hi", "hu", "hy", "id", "it", "ja", "ka", 
  "ko", "lt", "lv", "ms", "pl", "ps", "pt", "ro", "ru", "si", "sk", "so", "sq", 
  "sr", "sw", "ta", "th", "tk", "tr", "uk", "ur", "uz", "vi", "zh", "zh-hk", 
  "zh-tw"
]
```